storage/google/cloud/storage/blob.py

```diff
@@ -380,6 +380,110 @@ def download_as_string(self, client=None):
         self.download_to_file(string_buffer, client=client)
         return string_buffer.getvalue()
 
+    def _create_upload(
+            self, client, file_obj=None, size=None, content_type=None,
+            chunk_size=None, strategy=None, extra_headers=None):
+        """Helper for upload methods.
+
+        Creates a :class:`google.cloud.core.streaming.Upload` object to handle
+        the details of uploading a file to Cloud Storage.
+
+        :type client: :class:`~google.cloud.storage.client.Client` or
+            ``NoneType``
+        :param client: Optional. The client to use.  If not passed, falls back
+            to the ``client`` stored on the blob's bucket.
+
+        :type file_obj: file
+        :param file_obj: A file handle open for reading.
+
+        :type size: int
+        :param size: The size of the upload, in bytes.
+
+        :type content_type: str
+        :param content_type: Optional type of content being uploaded.
+
+        :type chunk_size: int
+        :param chunk_size: The size of each chunk when doing resumable and
+            media uploads.
+
+        :type strategy: str
+        :param strategy: Either
+            :attr:`google.cloud.core.streaming.transfer.SIMPLE_UPLOAD` or
+            :attr:`google.cloud.core.streaming.transfer.RESUMABLE_UPLOAD`.
+
+        :type extra_headers: dict
+        :param extra_headers: Additional headers to be sent with the upload
+            initiation request.
+
+        :rtype: Tuple[google.cloud.core.streaming.Upload,
+                      google.cloud.core.streaming.Request,
+                      google.cloud.core.streaming.Response]
+        :returns: The Upload object, the upload HTTP request, and the upload
+                  initiation response.
+        """
+
+        client = self._require_client(client)
+
+        # Use ``_base_connection`` rather ``_connection`` since the current
+        # connection may be a batch. A batch wraps a client's connection,
+        # but does not store the ``http`` object. The rest (API_BASE_URL and
+        # build_api_url) are also defined on the Batch class, but we just
+        # use the wrapped connection since it has all three (http,
+        # API_BASE_URL and build_api_url).
+        connection = client._base_connection
+
+        content_type = (content_type or self._properties.get('contentType') or
+                        'application/octet-stream')
+
+        headers = {
+            'Accept': 'application/json',
+            'Accept-Encoding': 'gzip, deflate',
+            'User-Agent': connection.USER_AGENT,
```

Comment:
Nope, these headers are for the initial request to start the upload which contains the blob metadata. The content type is `application/json` for this request. The `Upload` object takes care of sending the right content type when uploading the actual file.